controllers/managedocs_controller.go

```diff
@@ -1094,38 +1094,17 @@ func (r *ManagedOCSReconciler) reconcileK8SMetricsServiceMonitorAuthSecret() err
 		if err := r.own(r.k8sMetricsServiceMonitorAuthSecret); err != nil {
 			return err
 		}
-
-		secret := &corev1.Secret{}
-		secret.Name = grafanaDatasourceSecretName
-		secret.Namespace = openshiftMonitoringNamespace
-		if err := r.unrestrictedGet(secret); err != nil {
-			return fmt.Errorf("Failed to get grafana-datasources secret from openshift-monitoring namespace: %v", err)
-		}
-
-		authInfoStructure := struct {
-			DataSources []struct {
-				BasicAuthPassword string `json:"basicAuthPassword"`
-				BasicAuthUser     string `json:"basicAuthUser"`
-			} `json:"datasources"`
-		}{}
-
-		if err := json.Unmarshal(secret.Data[grafanaDatasourceSecretKey], &authInfoStructure); err != nil {
-			return fmt.Errorf("Could not unmarshal Grapana datasource data: %v", err)
-		}
-
-		r.k8sMetricsServiceMonitorAuthSecret.Data = nil
-		for key := range authInfoStructure.DataSources {
-			ds := &authInfoStructure.DataSources[key]
-			if ds.BasicAuthUser == "internal" && ds.BasicAuthPassword != "" {
-				r.k8sMetricsServiceMonitorAuthSecret.Data = map[string][]byte{
-					"Username": []byte(ds.BasicAuthUser),
-					"Password": []byte(ds.BasicAuthPassword),
-				}
+		var auth map[string][]byte = nil
+		var err error = nil
+		if auth, err = r.readGrafanaV1Secret(); err != nil {
+			if errors.IsNotFound(err) {
+				auth, err = r.readGrafanaV2Secret()
+			}
+			if err != nil {
+				return fmt.Errorf("Unable to find grafana-datasources secret: %v", err)
```

Comment:
You already know that `err != nil` from line 1099
```suggestion
			return fmt.Errorf("Unable to find grafana-datasources secret: %v", err)
```